src/niamoto/core/components/exports/page_generator.py

```
@@ -686,3 +687,377 @@
+                depth="../",
+                context={"taxa_data": taxa_data},
+            )
+
+            return rendered_path
+
+        except Exception as e:
+            raise TemplateError(
+                "taxon_index.html",
+                f"Failed to generate taxon index page: {str(e)}",
+                details={"error": str(e)},
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)